test/integration/serviceaccount/legacy_service_account_token_clean_up_test.go

```diff
@@ -133,19 +121,38 @@ func TestLegacyServiceAccountTokenCleanUp(t *testing.T) {
 	}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+
+			fakeClock := testingclock.NewFakeClock(time.Now().UTC())
+
+			// start legacy service account token cleaner
+			ctxForCleaner, cancelFunc := context.WithCancel(context.Background())
+			startLegacyServiceAccountTokenCleaner(ctxForCleaner, c, fakeClock, informers)
+			informers.Start(ctx.Done())
+			defer cancelFunc()
```

Comment:
I find this a big complex to understand 🤔

Reply:
Hi @aojea , is that because I use another `ctxForCleaner`? I want to create the cleaner for each test case and let it stop after the test finishes. So I don't want to use the common `ctx` to create the cleaner. Maybe this is not the usual way to do it. Do you have any suggestion?

Reply:
is not possible to reuse the NewLegacySATokenCleaner for all tests?

Reply:
oh, I see https://github.com/kubernetes/kubernetes/pull/119579/files#r1276497909

Reply:
I think that you don't need to start the informer, it seems is already started by `startServiceAccountTestServerAndWaitForCaches` 

https://github.com/kubernetes/kubernetes/blob/e68629e7b37acb1264f8518192b2d25b723d3be1/test/integration/serviceaccount/service_account_test.go#L429
```suggestion
			ctxForCleaner, cancelFunc := context.WithCancel(context.Background())
			startLegacyServiceAccountTokenCleaner(ctxForCleaner, c, fakeClock, informers)
			defer cancelFunc()
```

Reply:
We need to start the informer again. As in `startServiceAccountTestServerAndWaitForCaches`, it only starts the serviceaccounts, secrets and namespace informers which have been registered before. But in the `startLegacyServiceAccountTokenCleaner`, I register another pod informer. The informers.Start() function only start those informers have been registered with the informer factory. So I have to call the informers.Start() again to make it start the pod informer after I create the cleaner.

Reply:
I see, I should have paid more attention :) 

Thanks for clarifying

Reply:
Thank you all the same for reviewing that, @liggitt , @aojea !